src/services/ghost/GhostJetbrainsBridge.ts

```diff
@@ -0,0 +1,371 @@
+		} as unknown as GhostContextProvider
+	}
+
+	/**
+	 * Serialize completion results to a format suitable for RPC response
+	 */
+	private serializeCompletionResult(
+		completions: vscode.InlineCompletionItem[] | vscode.InlineCompletionList | undefined,
+		requestId: string,
+	): CompletionResult {
+		const items = Array.isArray(completions) ? completions : completions?.items || []
+
+		return {
+			requestId,
+			items: items.map((item) => ({
+				insertText: typeof item.insertText === "string" ? item.insertText : item.insertText.value,
+				range: item.range
+					? {
+							start: {
+								line: item.range.start.line,
+								character: item.range.start.character,
+							},
+							end: { line: item.range.end.line, character: item.range.end.character },
+						}
+					: null,
+			})),
+			error: null,
+		}
+	}
+
+	public async getInlineCompletions(...args: any[]): Promise<CompletionResult> {
+		try {
+			// Parse and validate arguments
+			const params = this.parseAndValidateArgs(...args)
+
+			// Normalize content
+			const { normalizedContent, lines } = this.normalizeContent(params.content)
+
+			// Determine language from languageId or file extension
+			const language = this.determineLanguage(params.languageId, params.uri)
+
+			// Create mock document
+			const mockDocument = this.createMockDocument(params.uri, normalizedContent, lines, language)
+
+			// Create VSCode position and context
+			const vscodePosition = new vscode.Position(params.position.line, params.position.character)
+			const context: vscode.InlineCompletionContext = {
+				triggerKind: vscode.InlineCompletionTriggerKind.Invoke,
+				selectedCompletionInfo: undefined,
+			}
+			const tokenSource = new vscode.CancellationTokenSource()
+
+			// Create mock context provider to prevent workspace file access
+			const mockContextProvider = this.createMockContextProvider(normalizedContent)
+
+			// Save original builders
+			const originalFimBuilder = this.ghost.inlineCompletionProvider.fimPromptBuilder
+			const originalHoleFiller = this.ghost.inlineCompletionProvider.holeFiller
+
+			try {
+				// Temporarily replace builders with ones using mock context
+				this.ghost.inlineCompletionProvider.fimPromptBuilder = new FimPromptBuilder(mockContextProvider)
+				this.ghost.inlineCompletionProvider.holeFiller = new HoleFiller(mockContextProvider)
+
+				// Get completions from the provider (will use mock builders internally)
+				const completions = await this.ghost.inlineCompletionProvider.provideInlineCompletionItems(
+					mockDocument,
+					vscodePosition,
+					context,
+					tokenSource.token,
+				)
+
+				// Serialize and return the result
+				return this.serializeCompletionResult(completions, params.requestId)
+			} finally {
+				// Always restore original builders
+				this.ghost.inlineCompletionProvider.fimPromptBuilder = originalFimBuilder
+				this.ghost.inlineCompletionProvider.holeFiller = originalHoleFiller
+				tokenSource.dispose()
+			}
```

Comment:
This is just a _liiiiitle_ gross 😬 But yeah, other than duplicating a lot of that inline completion provider, I can't really think of a _shorter_ solution than this one.

Since this code will ever only spin up internally as part of JetBrains in that build, do we ever really need to set it back to the original or is that just for cleanliness? 
```
				// Always restore original builders
				this.ghost.inlineCompletionProvider.fimPromptBuilder = originalFimBuilder
				this.ghost.inlineCompletionProvider.holeFiller = originalHoleFiller
```

Reply:
This command it's only available on jetbrains, I restored the override to be explicit on the temporary override pattern